script.js

```diff
@@ -1,12 +1,602 @@
+  const freeSearchResultShowsElem = document.getElementById('freeSearchResultShows');
+  let freeSearchValue = e.target.value.toLowerCase();
+  const filteredShows = allShows.filter(show => { 
+    let showName = show.name.toLowerCase();
+    let showGenres = show.genres.join(" ").toLowerCase();
+    let showSummary = show.summary.toLowerCase();
+
+    if (showName.includes(freeSearchValue) || showGenres.includes(freeSearchValue) || showSummary.includes(freeSearchValue)) {
+      return show;
+    };
+  });
+
+  console.log(filteredShows);
+  if (freeSearchValue == ""){
+    freeSearchResult.style.display = 'none';
+  } else if (filteredShows == "") {
+    freeSearchResult.style.display = 'flex';
+    freeSearchResultShowsElem.innerHTML = `<p>No shows found </br></p>`;
+  } else {
+    freeSearchResultShows(filteredShows);
+  }
+  
+  return freeSearchValue;
+});
+
+function freeSearchResultShows(shows){
+  const freeSearchResult = document.getElementById('freeSearchResult');
+  freeSearchResult.style.display = 'flex';
+  const freeSearchResultShows = document.getElementById('freeSearchResultShows');
+
+  let freeSearchResultShowsHtml = shows.map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  freeSearchResultShows.innerHTML = freeSearchResultShowsHtml;
+
+}
+
+const changeFormSection = () => {
+  if (freeSearchCont.style.display === 'flex' && seasonInputContainer.style.display === 'none') {
+    freeSearchCont.style.display = 'none';
+    allShowsContainer.style.display = 'none';
+    castListSection.style.display = 'flex';
+    seasonInputContainer.style.display = 'flex';
+  }
+  freeSearchCont.style.display = 'flex';
+  castListSection.style.display = 'none';
+  allShowsContainer.style.display = 'block';
+  seasonInputContainer.style.display = 'none';
+};
+
+const getGenreListing = () => {
+  let genresList = [];
+  let allGenres = [];
+  let showGenresObject = {};
+
+  for (let i = 0; i < allShows.length; i++){
+    let showGenres = allShows[i].genres;
+    for (let j = 0; j < showGenres.length; j++){
+      if (!genresList.includes(showGenres[j])){
+        let currentGenre = showGenres[j];
+        
+        genresList.push(showGenres[j])
+        showGenresObject[`${currentGenre}`] = allShows.filter((show) => {
+          if(show.genres.includes(currentGenre)){
+            return show
+          }
+        });
+      }
+    }
+  }
+  allGenres.push(showGenresObject);
+  return allGenres.sort();
```

Comment:
Perhaps here you could just return the showGenresObject (not return an array of size 1).

That way, when you use it lower down, you won't need to use [0] to access the object.

Reply:
Thanks Laura, done some rework on this and the destructuring comment you left. Returned showGenreObject destructing to the allGenres array so I can manipulate it better in the onLoadScreen Function and condense everything thanks for that.